crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -671,9 +671,12 @@ where
                         warn!(target: "consensus::engine", %err, "Failed to get finalized block header");
                     }
                     Ok(None) => {
-                        // we don't have the block yet and the distance exceeds the allowed
-                        // threshold
-                        return Some(state.finalized_block_hash)
+                        // ensure the finalized block is known (not the zero hash)
+                        if !state.finalized_block_hash.is_zero() {
+                            // we don't have the block yet and the distance exceeds the allowed
+                            // threshold
+                            return Some(state.finalized_block_hash)
```

Comment:
what happens when the distance is large and we return None here? do we just keep downloading?

Reply:
yeah we will keep trying to close the gap to head, we cant start pipeline sync if we don't have a finalized block